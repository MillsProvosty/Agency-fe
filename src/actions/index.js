export const setRole = role => ({
  type: 'SET_ROLE',
  role
});

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const setUserOpportunities = opportunities => ({
  type: 'SET_USER_OPPORTUNITIES',
  opportunities
});

export const addOpp = opp => ({
  type: 'ADD_OPP',
  opp
});

export const editOpp = opp => ({
  type: 'EDIT_OPP',
  opp
});
