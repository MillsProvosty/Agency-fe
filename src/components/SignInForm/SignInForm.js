import React, { Component } from "react";
import "./SignInForm.scss";
import { FaHandsHelping } from "react-icons/fa";
import { Route, Link } from "react-router-dom";
import Profile from "../Profile/Profile";
import { SignUpModal } from "../SignUp/SignUpModal";
import Modal from "react-modal";
import styled from "styled-components";

const SignIn = styled.section`
  height: auto;
  background-color: white;
  justify-content: center;
`;

const ModalStyle = styled.section`
  margin-top: 50px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const Container = styled.section`
  width: 100%;
`;

const Titles = styled.section`
  display: flex;
  @media screen and (max-width: 375px) {
    display: ${props => (props.second ? 'block': 'flex')}
    margin: ${props => (props.second ? '0px 50%': 'none')}
  }
  @media screen and (display-mode: standalone) {
    display: block;
    margin: 0px 50%;
  }

`;

const SignsSection = styled.section`
  width: 50%;
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 20vh;
  border-right: ${props => (props.sign ? "1px solid darkblue" : "none")};
  @media screen and (max-width: 375px) {
    border-right: none;
  }
  @media screen and (display-mode: standalone) {
    border-right: none;
  }
`;

const Headers = styled.h1`
  font-size: 6em;
  margin: 0px;
  display: flex;
  align-content: flex-start;
  @media screen and (max-width: 375px) {
    font-size: 3rem;
    padding: 0px 5px;
    margin-bottom: 20px;
    display: ${props => (props.SignUp ? 'none': 'flex')}
  }
  @media screen and (display-mode: standalone) {
    font-size: 3rem;
    padding: 0px 5px;
    margin-bottom: 20px;
    display: ${props => (props.SignUp ? 'none': 'flex')}
  }
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  @media screen and (max-width: 375px) {
    border-bottom: 2px solid darkblue;
    padding-bottom: 30px;
    margin-top: 20px;
  }
  @media screen and (display-mode: standalone) {
    border-bottom: 2px solid darkblue;
    padding-bottom: 30px;
    margin-top: 20px;
  }

`;

const Input = styled.input`
  display: block;
  width: 175px;
  margin-bottom: 10px;
  border-radius: 5px;
  font-size: 1em;
  height: 2em;
  border: 1px solid darkgrey;
  padding: 5px;
`;

const Button = styled.button`
  color: white;
  background-color: darkblue;
  border-radius: 5px;
  font-size: 2em;
  width: 175px;
  font-family: "Quicksand", sans-serif;
  border: 2px solid white;

  :hover {
    border: 2px solid darkblue;
    color: darkblue;
    background-color: white;
  }

  @media screen and (max-width: 375px) {
    margin-top: 10px
  }

  @media screen and (display-mode: standalone) {
    margin-top: 10px
  }
`;

export class SignInForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
      modalIsOpen: false
    };
  }

  displayModal = () => {
    this.setState({ modalIsOpen: true });
  };

  displayProfile = e => {
    e.preventDefault();
    return (
      <main>
        <Route
          exact
          path="/profile"
          render={() => <Profile props={"user info here"} />}
        />
      </main>
    );
  };

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    return (
      <SignIn className="SignInForm">
        <ModalStyle className="modal">
          <Modal
            isOpen={this.state.modalIsOpen}
            displayProfile={this.displayProfile}
            className="react-modal"
          >
            <SignUpModal />
          </Modal>
        </ModalStyle>
        <FaHandsHelping size={64} />
        <Container className="container">
          <Titles className="titles">
            <Headers>Sign In</Headers>
            <Headers SignUp>Sign Up</Headers>
          </Titles>
          <Titles second className="forms">
            <SignsSection sign className="sign-in">
              <Form>
                <Input
                  tabIndex={0}
                  type="text"
                  value={this.state.email}
                  name="email"
                  placeholder="example@email.com"
                  onChange={this.handleChange}
                />
                <Input
                  tabIndex={0}
                  type="text"
                  value={this.state.password}
                  name="password"
                  placeholder="Password"
                  onChange={this.handleChange}
                />
                <Link
                  to="/profile"
                  tabIndex={0}
                  style={{ textDecoration: "none" }}
                >
                  <Button>Sign In</Button>
                </Link>
              </Form>
            </SignsSection>
            <SignsSection className="sign-up">
              <Button onClick={this.displayModal}>Sign Up</Button>
            </SignsSection>
          </Titles>
        </Container>
      </SignIn>
    );
  }
}
