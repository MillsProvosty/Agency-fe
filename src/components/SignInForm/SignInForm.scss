.SignInForm {
    height: auto;
    background-color: white;
    // border: 1px solid lightgrey;
    // border-radius: 8px;
    // display: flex;
    // flex-direction: column;
    // width: 500px;
    // padding: 20px;
    justify-content: center;

    & .container{
        display: flex;

    }

    & .sign-in, .sign-up {
        width: 50%;
        display: flex;
        flex-direction: column;
        align-items: center;
        height: 40vh;
        
        & form {
            justify-content: space-around;

            & input, button {
                justify-content: space-around;
                display: block;
                width: 175px;
            }

            & input {
                border-radius: 5px;
                font-size: 1em;
                height: 2em;
            }


        }
    }
}