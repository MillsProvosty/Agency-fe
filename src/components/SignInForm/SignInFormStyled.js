
import styled from "styled-components";
import floatingImg from "./floating.svg";

export const SignIn = styled.section`
  justify-content: center;
  height: 100vh;
  width: 100%;
  background-color: aliceblue;
  background-repeat: no-repeat;
  background-size: 1000px;
  background-position: center 10vh;
  background-image: url(${floatingImg});
`;

export const ModalStyle = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const Container = styled.section`
  width: 100%;
`;

export const Titles = styled.section`
  display: flex;
  @media screen and (max-width: 375px) {
    display: ${props => (props.second ? "block" : "flex")}
    margin: ${props => (props.second ? "0px 50%" : "none")}
  }
  @media screen and (display-mode: standalone) {
    display: block;
    margin: 0px 50%;
  }
`;

export const SignsSection = styled.section`
  width: 40%;
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 43vh;
  margin-top: 27vh;
  background-color: white;
  border: 1px solid #37474e;
  -webkit-box-shadow: 9px 10px 5px -10px rgba(55, 71, 78, 1);
  -moz-box-shadow: 9px 10px 5px -10px rgba(55, 71, 78, 1);
  box-shadow: 9px 10px 5px -10px rgba(55, 71, 78, 1);
  border: 1px solid #37474e @media screen and (max-width: 375px) {
    border-right: none;
  }
  @media screen and (display-mode: standalone) {
    border-right: none;
  }
`;

export const TitleSection = styled.section`
  display: flex;
  background-color: aliceblue;
  margin-top: 23vh;
  align-items: center;
  width: 80px;
  margin-left: 5%;
`;

export const Headers = styled.h1`
  font-size: ${props => (props.title ? "3em" : "6em")};
  margin: 0px;
  display: flex;
  align-content: flex-start;
  @media screen and (max-width: 375px) {
    font-size: 3rem;
    padding: 0px 5px;
    margin-bottom: 20px;
    display: ${props => (props.SignUp ? "none" : "flex")};
  }
  @media screen and (display-mode: standalone) {
    font-size: 3rem;
    padding: 0px 5px;
    margin-bottom: 20px;
    display: ${props => (props.SignUp ? "none" : "flex")};
  }
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  margin: auto;
  @media screen and (max-width: 375px) {
    border-bottom: 2px solid #37474e;
    padding-bottom: 30px;
    margin-top: 20px;
  }
  @media screen and (display-mode: standalone) {
    border-bottom: 2px solid #37474e;
    padding-bottom: 30px;
    margin-top: 20px;
  }
`;

export const Input = styled.input`
  display: block;
  width: 175px;
  margin-bottom: 10px;
  border-radius: 5px;
  font-size: 1em;
  height: 2em;
  border: 1px solid darkgrey;
  padding: 5px;
`;


export const PTag = styled.p`
  font-size: 1em;
  color: tomato;
  margin: 0;
`;

export const Logo = styled.button`
  background-color: white;
  border-radius: 50%;
  border: 1px solid #37474e;
`;

export const Button = styled.button`
  color: white;
  background-color: #7a86cb;
  border-radius: 5px;
  font-size: 2em;
  width: 175px;
  font-family: "Quicksand", sans-serif;
  border: 2px solid white;

  :hover {
    border: 2px solid #7a86cb;
    color: #7a86cb;
    background-color: white;
  }

  @media screen and (max-width: 375px) {
    margin-top: 10px;
  }

  @media screen and (display-mode: standalone) {
    margin-top: 10px;
  }
`;