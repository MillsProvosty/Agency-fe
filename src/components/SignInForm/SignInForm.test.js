import React from 'react';
import { shallow } from 'enzyme';
import { SignInForm, mapDispatchToProps } from './SignInForm';
import { setUser } from '../../actions';

describe('SignInForm', () => {
  let wrapper;

  const mockUser = {
    id: 1,
    role: 'client',
    name: 'wang'
  }
  const mockSetUser = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
      <SignInForm 
        setAUser={mockSetUser}
      />
    )
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('button should be disabled onclick', () => {
    wrapper.find('#sign-in').props().onClick();
    expect(mockSetUser).not.toHaveBeenCalled();
  })

  // it('should call setUser', () => {

  // })
  describe('mapDispatchToProps', () => {
    it("calls setAUser with an setUser action when setAUser is called", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setUser(mockUser);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setAUser(mockUser);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
})