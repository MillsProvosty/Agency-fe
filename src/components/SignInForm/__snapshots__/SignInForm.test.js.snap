// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm should match snapshot 1`] = `
<styled.section>
  <styled.section>
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="modal"
      closeTimeoutMS={0}
      isOpen={false}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <Connect(SignUpModal) />
    </Modal>
  </styled.section>
  <styled.section>
    <styled.section
      second={true}
    >
      <styled.section
        sign={true}
      >
        <styled.form>
          <styled.h1>
            Sign In
          </styled.h1>
          <styled.input
            autoComplete="off"
            name="email"
            onChange={[Function]}
            placeholder="example@email.com"
            required={true}
            tabIndex={0}
            type="email"
            value=""
          />
          <styled.input
            autoComplete="off"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            tabIndex={0}
            type="text"
            value=""
          />
          <NavLink
            style={
              Object {
                "textDecoration": "none",
              }
            }
            tabIndex={0}
            to="/profile"
          >
            <styled.button
              disabled={true}
              id="sign-in"
              onClick={[Function]}
            >
              Sign In
            </styled.button>
          </NavLink>
        </styled.form>
      </styled.section>
      <styled.section>
        <styled.h1
          SignUp={true}
        >
          Sign Up
        </styled.h1>
        <styled.button
          onClick={[Function]}
        >
          Sign Up
        </styled.button>
      </styled.section>
    </styled.section>
    <styled.section
      header={true}
    >
      <styled.button>
        <GiAirBalloon
          className="hands"
          size={40}
          style={
            Object {
              "color": "#37474e",
            }
          }
        />
      </styled.button>
    </styled.section>
  </styled.section>
</styled.section>
`;
